sentry/src/main/java/io/sentry/metrics/Metric.java

```diff
@@ -0,0 +1,66 @@
+package io.sentry.metrics;
+
+import io.sentry.MeasurementUnit;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Base class for metric instruments */
+@ApiStatus.Internal
+public abstract class Metric {
+
+  private final @NotNull String key;
+  private final @Nullable MeasurementUnit unit;
+  private final @Nullable Map<String, String> tags;
+  private final @NotNull Long timestampMs;
+
+  /**
+   * Creates a new instance of {@link Metric}.
+   *
+   * @param key The text key to be used to identify the metric
+   * @param unit An optional {@link MeasurementUnit} that describes the values being tracked
+   * @param tags An optional set of key/value pairs that can be used to add dimensionality to
+   *     metrics
+   * @param timestampMs A time when the metric was emitted.
+   */
+  public Metric(
+      @NotNull String key,
+      @Nullable MeasurementUnit unit,
+      @Nullable Map<String, String> tags,
+      @NotNull Long timestampMs) {
```

Comment:
you can add the type here, so you don't have to override the getType() everywhere

Reply:
Yeah, I wanted to save some per-instance memory here 😅 , but fine for me.